src/lib/tvgSwCanvas.cpp

```diff
@@ -67,6 +67,12 @@ Result SwCanvas::target(uint32_t* buffer, uint32_t stride, uint32_t w, uint32_t
 
     if (!renderer->target(buffer, stride, w, h, cs)) return Result::InvalidArguments;
 
+    Array<RenderData> clips;
```

Comment:
how about this? this could supply both gl, sw engines. also more efficient than your version.

SwCanvas::target() {
   ...
   Canvas::pImpl->invalidate();
}

Canvas::Impl::invalidate()
{
   foreach(...) paint->pImpl->flags = RenderUpdate::All
}